pymongo/topology.py

```diff
@@ -237,6 +240,9 @@ def select_server(self,
                       server_selection_timeout=None,
                       address=None):
         """Like select_servers, but choose a random server if several match."""
+        if self._is_closed:
+            raise InvalidOperation("Once a Topology is closed, "
+                                   "all operations raise an error")
```

Comment:
I suggest moving this to the Topology._ensure_opened for a few reasons:
- this is the only way to ensure that we won't re-open the topology.
- we should be holding the lock when checking _is_closed.

Reply:
That makes sense. Fixed.